### ISSUE TITLE
Why cv::resize with CV_8S and CV_32S type was disabled?

### ISSUE THREAD

**Issue by logic1988** [opencv/opencv#7862]
Using cv::resize function with cv::Mat in `CV_8S` and `CV_32S` type will cause an error:
`opencv/modules/imgproc/src/imgwarp.cpp:3368: error: (-215) func != 0 in function resize`
After review of the source code I found:
```
    static ResizeFunc linear_tab[] =
    {
        resizeGeneric_<
            HResizeLinear<uchar, int, short,
                INTER_RESIZE_COEF_SCALE,
                HResizeLinearVec_8u32s>,
            VResizeLinear<uchar, int, short,
                FixedPtCast<int, uchar, INTER_RESIZE_COEF_BITS*2>,
                VResizeLinearVec_32s8u> >,
        0,
        resizeGeneric_<
            HResizeLinear<ushort, float, float, 1,
                HResizeLinearVec_16u32f>,
            VResizeLinear<ushort, float, float, Cast<float, ushort>,
                VResizeLinearVec_32f16u> >,
        resizeGeneric_<
            HResizeLinear<short, float, float, 1,
                HResizeLinearVec_16s32f>,
            VResizeLinear<short, float, float, Cast<float, short>,
                VResizeLinearVec_32f16s> >,
        0,
        resizeGeneric_<
            HResizeLinear<float, float, float, 1,
                HResizeLinearVec_32f>,
            VResizeLinear<float, float, float, Cast<float, float>,
                VResizeLinearVec_32f> >,
        resizeGeneric_<
            HResizeLinear<double, double, float, 1,
                HResizeNoVec>,
            VResizeLinear<double, double, float, Cast<double, double>,
                VResizeNoVec> >,
        0
    };
```
This array corresponds to the type of Mat:
```
#define CV_8U   0
#define CV_8S   1
#define CV_16U  2
#define CV_16S  3
#define CV_32S  4
#define CV_32F  5
#define CV_64F  6
#define CV_USRTYPE1 7
```
The array value associated with `CV_8S` and `CV_32S` is 0, which led to error.
`cv::resize` is an important function, but the bug seems to exist for a long time, which have any special reasons?

**Comment by alalek** [comment]
Do you have real use cases for these formats, especially with linear interpolation?
There are more relevant formats:
32S -> 32F
8S -> 16S

**Comment by logic1988** [comment]
Yes, but I am very surprised why 16S can, but 8S and 32S but not.

**Comment by HGuillemet** [comment]
`CV_32S` and `CV_8S` are supported by opencv and geometric transformations are quite basic features everyone uses.
Of course there are use cases where we want to rescale 32S or 8S images.
Would it really clutter the library too much to add support for these types ?
Please consider adding this support.

**Comment by MK-3PP** [comment]
> Do you have real use cases for these formats, especially with linear interpolation? There are more relevant formats: 32S -> 32F 8S -> 16S

Yes.

As is, I need to actively work around the corner cases, sacrificing ressources at runtime (temporary memory, conversion time) and development time. **Plus, I need to make sure my workaround does not introduce silly new rounding or overflow errors** (that are inconsistent with OpenCV's).
Alternatively, I could forward the issue to my users and get some jimmies rustled.

**Comment by emmenlau** [comment]
We are using OpenCV to build a higher-level image analysis framework, and cases like this make it really cumbersome. In the higher levels algorithm development, you just need to rely on lower levels working, without having to remember myriads of workarounds. Would it be possible to fix this at the source?
